TSOClient/tso.simantics/NetPlay/EODs/Utils/EODLobby.cs

```diff
@@ -95,10 +95,20 @@ public bool IsFull()
             return true;
         }
 
+        public bool IsEmpty()
+        {
+            for (var i = 0; i < Players.Length; i++)
+            {
+                if (Players[i] != null)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
         public bool Join(VMEODClient client, short slot)
         {
-            if (slot == -1) //take first available slot
```

Comment:
This is necessary for trade...

Reply:
For the EODLobby.cs change, I think I just missed that it had been updated for Secure Trade and was working on the older version. Apologies and thanks for patching it.